### ISSUE TITLE
[Request] for addition of Mangadex

### ISSUE THREAD

**Issue by auz-jsm** [raulhaag/MiMangaNu#620]
Please, well if possible, allow to include Mangadex.org in the server list. Thanks!

**Comment by raulhaag** [comment]
yes, look like batoto kind of server, no time for now but i will take in account.

**Comment by aelisya** [comment]
i have asked them if they had an api, but this one is very very little (not enought for using it for this kind of application, and the rss isn't complete on most of their manga).

So it would be VERY difficult to implement.

**Comment by asdkant** [comment]
Tachiyomi has a mangadex plugin, if you open the page in web view once to pass the DDoS check it works fine, you could imitate how that one works.

**Comment by borrougagnou** [comment]
use MiMangaNu with mangadex are difficult because Mangadex:

**1) work with multiple language...**
and when you have this: 
![image](https://user-images.githubusercontent.com/10796546/61718991-55d70300-ad64-11e9-924b-ca52e0d1ee9b.png)

it's difficult to short all of that...

**2) the scan of each new chapter will take a long time ... (because multiple parse)**

**Comment by Fraulk** [comment]
1) Batoto was on the app before dying, and there were multiple languages so it's surely not a problem
2) You can choose fast scan to check only the last 20 chapters, or be patient

**Comment by borrougagnou** [comment]
1) I'm not here at the beginning but if batoto disapear on the list, I think its because the MiMangaNu app didn't have lots of functionality that we have now
+ batoto separate correctly each language from the list by URL ... very different from mangadex who each chapter are mixed with other for one url...

2) fast scan to check only 20th chapter... lul... didn't change nothing because you always need a lots of short...
like this... 
![image](https://user-images.githubusercontent.com/10796546/61722402-8d48ae00-ad6a-11e9-95aa-c7c94a4eefb7.png)
during the 20th chapter you have only EN + BR language, the others language are below...
as I said, lots of sort...

**Comment by Fraulk** [comment]
Mangadex is a huge server and there's a lot of manga, frequently updated by the teams, so why are you against it ? Yeah it could be difficult to implement but it's definitely worth it

**Comment by auz-jsm** [comment]
I'm no a programmer (so idk the complexity it might dealt) but Mangadex.org has actually a language sorter (in their search results and in your account's language preference)

**Comment by borrougagnou** [comment]
I know that mangadex is a huge server with lots of manga
the problem is "implementation" into MiMangaNu...
because the stupid devs of Mangadex didn't code the site properly !

but wait, maybe I found something who was helpful

**Comment by aelisya** [comment]
I'm against it too, because i would be using a lot a offort for a server who is actually in difficulty, if they don"t do something they risque to die ?

Why ? because a lot of manga aren't updated anymore by the team who have quit the boat, for exemple https://mangadex.org/title/24213/29-to-jk (the team have update it since, but not on mangadex, and mangadex remove the content if the team ask for so the user canot update it manually), i don't use mangadex anymore because of that.

So explain to us, why us (developer), have to work hours, for a server who isn't up to date ?
(it's a real question not a troll).

**Comment by aelisya** [comment]
plus they have an api, but it's very limited and not documented, so it will be lot of effort for use this api (and one update and poof api dead).

**Comment by borrougagnou** [comment]
Ha, yes I forgot about that...

I though use api but they dont have any list just info about eps...
And if they prepare a new interface, it will stupid to begin something...

So, it's close for myself

**Comment by aelisya** [comment]
Yes the ideal is if the open a public api for retrieve last manga up, and research api, but since none of the source want to do it we are lock to use html read

**Comment by raulhaag** [comment]
hi, i found 2 usefull parts in the api the manga and chapter, there are fast and complete as i see in response, extra config go to be needed to add to it the lengauges codes to filter, also saomething like chapter collapse for the chapter with multiple scalation (i see this part as the problematic but can be aplyed to other servers) or keep all scalation but not usefull at read moment one after other repeating

**Comment by aelisya** [comment]
we already found it, (but they're going to make changes that might break the api)

**Comment by aelisya** [comment]
Confirmation from mangadex V5 on the Way
full API will be available in this update
https://mangadex.org/thread/195435/1

**Comment by raulhaag** [comment]
Hi @aelisya I started a branch a time ago and i was working on it but, not to many time and y forget about it for a couple of month, if you want to test it and i pushed this branch
https://github.com/raulhaag/MiMangaNu/tree/Mangadex

**Comment by aelisya** [comment]
i will check on that :)
